python3.12/apigw-pytorch/{{cookiecutter.project_name}}/app/Dockerfile

```diff
@@ -0,0 +1,8 @@
+FROM public.ecr.aws/lambda/python:3.12
```

Comment:
nit: Not a blocker, could also be done in a separate PR but pasting here for reference. We can change the Dockerfiles for consistency with the changes in this [PR](https://github.com/aws/aws-sam-cli-app-templates/pull/461/files)